### PR TITLE
Canned food interaction update

### PR DESCRIPTION
**Pull request by quardbreak**
- Added can-opener knife, a knife subtype.
- You can't open cans with bare hand now. You need to use knife to open it.
- - Can-opener require 5 seconds to open can. Regular knife require 15 seconds.
- You can feed people with can and use fork on it.

### CODE REVIEW

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	if(!sealed)
+		unseal()
+
+/obj/item/chems/food/snacks/can/examine(mob/user)
+	. = ..()
+	to_chat(user, "It is [sealed ? "" : "un"]sealed.")
+	to_chat(user, "It looks [open_complexity ? "hard" : "easy "] to open.")
+
+/obj/item/chems/food/snacks/can/proc/unseal(mob/user)
+	playsound(src, 'sound/effects/canopen.ogg', rand(10, 50), 1)
```
Comment (afterthought2) [inline code/modules/reagents/reagent_containers/food/canned_food.dm:27]: The volume is randomized? Why?

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		return TRUE
+	return ..()
+
+/obj/item/chems/food/snacks/can/attack_self(mob/user)
+	if(!ATOM_IS_OPEN_CONTAINER(src) && sealed && !open_complexity)
+		to_chat(user, SPAN_NOTICE("You unseal \the [src] with a crack of metal."))
+		unseal()
+
+/obj/item/chems/food/snacks/can/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/knife) && sealed)
```
Comment (afterthought2) [inline code/modules/reagents/reagent_containers/food/canned_food.dm]: Isn't there a fancy system for tools like this now?

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	else if(istype(W,/obj/item/kitchen/utensil))
+		if(ATOM_IS_OPEN_CONTAINER(src))
+			..()
+		else
+			to_chat(user, SPAN_WARNING("You need a can-opener to open this!"))
+	return
+
+/obj/item/chems/food/snacks/can/on_update_icon()
+	if(!sealed)
+		icon_state = "[initial(icon_state)]-open"
```
Comment (afterthought2) [inline code/modules/reagents/reagent_containers/food/canned_food.dm:73]: And if it is sealed?

Comment (quardbreak) [reply code/modules/reagents/reagent_containers/food/canned_food.dm:73]: ? The sprite doesn't change self if it not unsealed

File: code/modules/reagents/reagent_containers/food/canned_food.dm (hunk 1 above)
Comment (quardbreak) [reply code/modules/reagents/reagent_containers/food/canned_food.dm:27]: Because it was as before

**Comment by quardbreak** [comment]
Thw entire file is a bit changed old file with the same code as before.

File: code/modules/reagents/reagent_containers/food/canned_food.dm (hunk 3 above)
Comment (MistakeNot4892) [reply code/modules/reagents/reagent_containers/food/canned_food.dm:73]: What if it is resealed via varedit or something like that?

Comment (quardbreak) [reply code/modules/reagents/reagent_containers/food/canned_food.dm:73]: Ah, this. I suppose it should check for a flag

File: code/modules/reagents/reagent_containers/food/canned_food.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/modules/reagents/reagent_containers/food/canned_food.dm]: Knives aren't considered tools currently.

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+//Just a short line of Canned Consumables, great for treasure in faraway abandoned outposts
+
+/obj/item/chems/food/snacks/can/beef
+	name = "quadrangled beefium"
+	icon_state = "beef"
+	desc = "Proteins carefully cloned from an extinct species of cattle in a secret facility on the outer rim."
+	trash = /obj/item/trash/beef
+	filling_color = "#663300"
+	center_of_mass = @"{'x':15,'y':9}"
```
Comment (MistakeNot4892) [inline code/modules/reagents/reagent_containers/food/canned_food.dm]: Every can sets center_of_mass to the same value, just move it onto the root type.

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+	nutriment_desc = list("beef" = 1)
+	bitesize = 3
```
Comment (MistakeNot4892) [inline code/modules/reagents/reagent_containers/food/canned_food.dm]: Likewise bitesize can be set on the root type.

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 6]
```diff
@@ -0,0 +1,172 @@
+//Canned Foods - crack open, eat.
+
+#define OPEN_HARD 1
+#define OPEN_EASY 0
+
+/obj/item/chems/food/snacks/can
+	name = "void can"
```
Comment (MistakeNot4892) [inline code/modules/reagents/reagent_containers/food/canned_food.dm]: `empty can` would make the type at least useful without seeming strange.

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+/obj/item/chems/food/snacks/can/attack_self(mob/user)
+	if(!ATOM_IS_OPEN_CONTAINER(src) && !open_complexity)
+		to_chat(user, SPAN_NOTICE("You unseal \the [src] with a crack of metal."))
+		unseal()
+
+/obj/item/chems/food/snacks/can/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/knife) && !ATOM_IS_OPEN_CONTAINER(src))
+		user.visible_message(
+			SPAN_WARNING("\The [user] is trying to open \the [src] with \the [W]!"),
+			SPAN_NOTICE("You starting open \the [src]!")
```
Comment (MistakeNot4892) [inline code/modules/reagents/reagent_containers/food/canned_food.dm]: `You start to open \the [src].`.

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+/obj/item/chems/food/snacks/can/attack_self(mob/user)
+	if(!ATOM_IS_OPEN_CONTAINER(src) && !open_complexity)
+		to_chat(user, SPAN_NOTICE("You unseal \the [src] with a crack of metal."))
+		unseal()
+
+/obj/item/chems/food/snacks/can/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/knife) && !ATOM_IS_OPEN_CONTAINER(src))
+		user.visible_message(
+			SPAN_WARNING("\The [user] is trying to open \the [src] with \the [W]!"),
```
Comment (MistakeNot4892) [inline code/modules/reagents/reagent_containers/food/canned_food.dm]: `SPAN_NOTICE`

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+			to_chat(user, SPAN_NOTICE("You unseal \the [src] with a crack of metal."))
+			unseal()
+			return
+
+	else if(istype(W,/obj/item/kitchen/utensil))
+		if(ATOM_IS_OPEN_CONTAINER(src))
+			..()
+		else
+			to_chat(user, SPAN_WARNING("You need a can-opener to open this!"))
+	return
```
Comment (MistakeNot4892) [inline code/modules/reagents/reagent_containers/food/canned_food.dm]: Superfluous return.

File: code/modules/reagents/reagent_containers/food/canned_food.dm [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+	atom_flags |= ATOM_FLAG_OPEN_CONTAINER
+	sealed = FALSE
+
+/obj/item/chems/food/snacks/can/attack(mob/M, mob/user, def_zone)
+	if(force && !(obj_flags & ITEM_FLAG_NO_BLUDGEON) && user.a_intent == I_HURT)
+		return ..()
+
+	if(standard_feed_mob(user, M))
+		update_icon(src)
+		return
```
Comment (MistakeNot4892) [inline code/modules/reagents/reagent_containers/food/canned_food.dm]: `return TRUE`

**Comment by MistakeNot4892** [comment]
Sadly the rule of beasts is 'git says you touched it, it's your problem'.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**